cmd/fastly/main.go

```diff
@@ -39,11 +38,118 @@ func main() {
 		out            io.Writer = common.NewSyncWriter(os.Stdout)
 	)
 
+	// We have to manually handle the inclusion of the verbose flag here because
+	// Kingpin doesn't evaluate the provided arguments until app.Run which
+	// happens later in the file and yet we need to know if we should be printing
+	// output related to the application configuration file in this file.
+	var verboseOutput bool
+	for _, seg := range args {
+		if seg == "-v" || seg == "-verbose" || seg == "--verbose" {
+			verboseOutput = true
+		}
+	}
+
+	// Extract a subset of configuration options from the local application directory.
+	var file config.File
+	err := file.Read(config.FilePath)
+	if err != nil {
+		if verboseOutput {
+			text.Output(out, `
+				We were unable to locate a local configuration file required to use the CLI.
+				We will create that file for you now.
+			`)
+			text.Break(out)
+		}
+
+		err := file.Load(config.RemoteEndpoint, httpClient)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1) // TODO: offer a clearer remediation step
+		}
+	}
+
+	// When the local configuration file is stale we'll need to acquire the
+	// latest version and write that back to disk. To ensure the CLI program
+	// doesn't complete before the write has finished, we block via a channel.
+	waitForWrite := make(chan bool)
+	wait := false
+
+	// Validate if configuration is older than its TTL
+	if check.Stale(file.CLI.LastChecked, file.CLI.TTL) {
+		if verboseOutput {
+			text.Warning(out, `
+Compatibility and versioning information for the Fastly CLI is being updated in the background.  The updated data will be used next time you execute a fastly command.
```

Comment:
General quesiton which applies to a couple of output messages in this PR: whats the reasoning for _"Compatibility and versioning information"_ versus simply _"Configuration"_?

Reply:
Discussion [here](https://github.com/fastly/cli/pull/187#discussion_r574444593) and [here](https://github.com/fastly/cli/pull/187#discussion_r574486737).

Reply:
That discussion is about the use of 'local application' - which I agree with. However, @triblondon's suggested edit of "compatibility and versioning information" seems strange to me and might not always be true in the future, where a generic "configuration" will.

Reply:
I don't love what I suggested here.  It was to solve the 'local application' problem, but 'configuration' is too broad imho.  I think we need to give the user a sense of what this data is for.

Reply:
I'm OK with this for now.

Reply:
Still find this weird 😬